pkg/services/app/add.go

```diff
@@ -493,6 +529,27 @@ func (a *App) commitAndPush(filters ...func(string) bool) error {
 	return nil
 }
 
+func (a *App) getRepoInfo(repoUrl string, gitProvider gitproviders.GitProvider) (*gitprovider.RepositoryInfo, error) {
```

Comment:
`getRepoInfo` and `getDefaultBranch` functions seem more go-git-providers related, I would suggest to move them to the `gitproviders` package. There is already a function named `GetRepoInfo` which receives `account-type`, `owner` and `repoName`, we could create another one with a slightly different name to reflect the case when receiving the repoURL, maybe GetRepoInfoFromURL or similar.

Reply:
Okay, that makes sense. I wasn't really sure how much we wanted to put specific use cases in the go-git-providers interface (vs a single `GetRepoInfo` function that we then use in various places). I'll make the change.

Reply:
Done